### ISSUE TITLE
Argument of type 'number' | 'string' is not assignable to overloaded function accepting 'number' or 'string

### ISSUE THREAD

**Issue by osi-oswald** [Microsoft/TypeScript#8350]
**TypeScript Version:** 

1.8.10

**Code**

```ts
declare function overload(param: number);
declare function overload(param: string);

let bar: number | string;
overload(bar); // Argument of type 'number' | 'string' is not assignable to parameter of type 'string'
```

**Expected behavior:**
Since the function 'overload' either accepts a 'number' or a 'string', the compiler should not complain.

**Actual behavior:** 
The compiler complains with: "Argument of type 'number' | 'string' is not assignable to parameter of type 'string'".

**Workaround:**
Of course I could always rewrite the declaration of 'overload' as:
```ts
declare function overload(param: number | string);
```

But usually these declarations are from external *.d.ts files which I do not want to touch if I don't have to.

**Comment by yortus** [comment]
Looks like a duplicate of #1805

**Comment by osi-oswald** [comment]
You are absolutely right regarding the duplicate
